### ISSUE TITLE
remove encode('utf-8')

### ISSUE THREAD

**Issue by razisayyed** [aptivate/django-sortable-listview#16]
```
parse_qs(query_string.encode('utf-8'))
```
returns a dict with both keys and values are bytes.
This will work fine in Python 2. but in Python 3 bytes can not be compared to strings, and thus query parameters that supposed to be removed will be ignored except if you add the following line to the view:
```
del_query_parameters = [b'page', b'sort']
```
I think it is safe to remove encode('utf-8') and just use:
```
parse_qs(query_string)
```

**Comment by martinburchell** [comment]
Thanks for reporting this. Want to submit a PR?

**Comment by razisayyed** [comment]
It will be a pleasure

**Comment by bw-aus** [comment]
Hello @razisayyed could you please read issue #10 to see why this was introduced (for Py2). There was a test case added then, too.

Could you please ensure your PR makes this library work for both Py2 and Py3, and that the tests (especially test_non_ascii_params_dont_throw_exception) pass for both versions? Thanks!

**Comment by bw-aus** [comment]
Would something like this be acceptable for both py2 and py3 support?

https://github.com/bw-aus/django-sortable-listview/commit/b1ffd41c92d6716d68b5965c7826e3a1892f637f

I confirm that the test passes for py2 and py3 locally with this change. I can make a PR if others agree. @razisayyed does this work for you or do you have something better? Ideally someone with a py3 Django proj should do a manual test on this, too...

**Comment by razisayyed** [comment]
I think your approach will work for both py2 and 3. Unfortunatelly, I do not have tome to test it manually at the moment.

**Comment by bw-aus** [comment]
I ran up a py2.7 and py3.5 version of the example_project (simple_blog) in the codebase and confirmed this issue with py3.

I also confirmed that tox was showing that the relevant test was failing for py3 (as well as some other failures/errors, addressed below).

Instead of making py2 users modify their code (e.g. u'' or unicode_literals) I think it's best to use **six** to handle the different cases, so I have introduced a dependency on six and implemented the if-statement for the encode.

I introduced two other test fixes (not related to this issue) so that tox was passing for all versions I could test with:
- There was an erroneous assert in test_get_sort_string_is_called
- 'mock' is not in unittest for python 3.3 so I added it to the requirements in tox.ini

I have confirmed that the new code passes all tox tests locally for all Python versions in tox.
See my gist https://gist.github.com/bw-aus/195c05ea4eb20273ee414bfc9c1254a8 for passing test confirmation. (Travis CI still shows a couple of errors but only due to missing Python versions there.)

I have confirmed _manually_ that the example_project (simple_blog) no longer exhibits the problem with parameter deletion on py3.

I decided to let aptivate do the version bump when ready, so I didn't include it.

I hope someone can review and try on a py3 project manually! Thanks very much, sorry for the initial regression, hope this helps.

**Comment by martinburchell** [comment]
Thanks all for your contributions - will aim to look at this soon

**Comment by JanMalte** [comment]
@martinburchell any news?

**Comment by birdsarah** [comment]
I have merged #18 which should solve this.
